app/components/text-block.js

```diff
@@ -14,29 +22,73 @@ const TextBlock = props => {
     subPoints = [],
     actionText = '',
     action, // text for an anchor or a function to put in onClick of a button
+    iconName = null, // A string name corresponding to an svgr component
+    side = 'left', // The side to show the icon, if provided.
     ...otherProps
   } = props
   const classes = useStylesFromThemeFunction()
-  return (
-    <div className={cx(classes.textBlock, classes[mode], className)} {...otherProps}>
-      <div className={classes.wrapper}>
-        {subject && <h2 className={classes.subject}>{subject}</h2>}
-        {description && <MarkDown className={classes.description}>{description}</MarkDown>}
-        {subPoints && (
-          <ul className={classes.subPoints}>
-            {subPoints.map(text => (
-              <li>{text}</li>
-            ))}
-          </ul>
-        )}
-        {actionText && (
-          <div className={classes.actionButton}>
-            <ActionButton action={action}>{actionText}</ActionButton>
-          </div>
-        )}
-      </div>
+
+  // Verify a valid side has been provided
+  if (side != 'left' && side != 'right') {
+    console.error(`${side} is not a valid argument. Must be either 'left' or 'right'.`)
+  }
+
+  // Section for grouping text-based elements
+  const textSection = (
+    // Shrink width to 75% to fit the icon if provided, else leave as-is
+    <div className={iconName ? classes.textSectionWithIcon : classes.textSectionNoIcon}>
+      {subject && <h2 className={classes.subject}>{subject}</h2>}
+      {description && <MarkDown className={classes.description}>{description}</MarkDown>}
+      {subPoints && (
+        <ul className={classes.subPoints}>
+          {subPoints.map(text => (
+            <li>{text}</li>
+          ))}
+        </ul>
+      )}
+      {actionText && (
+        <div className={classes.actionButton}>
+          <ActionButton action={action}>{actionText}</ActionButton>
+        </div>
+      )}
     </div>
   )
+
+  // Check if the icon with name exists in svgr
+  let iconComponent = null
```

Comment:
This all works and its great but wouldn't this be a computationally quicker way than Object.keys:
```
const iconComponent= iconName && icons[iconName] && <Iconify iconName={iconName} width="10rem" height="auto" /> 
```
Also, instead of if (iconName != null) I just say if (iconName) or if(!iconName)  and you don't even have to initialize it to null at the top.   In general I would use null for objects, and '' for strings, but undefined is a thing in js  and always tests false.

Reply:
Ah yeah, not sure why I didn't just access those directly. Thanks!